### PR TITLE
average_of_integers

### PR DESCRIPTION
**Pull request by Mithchell509**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Review by Djokwa — DISMISSED** [review]
Everything is correct. Nicely done.

File: solutions/average_of_integers.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+"""
```
Comment (Alemayehu-Desta) [inline solutions/average_of_integers.py:1]: add Python interpreter lines at the top of the file

#!/usr/bin/env python3
# -*- coding: utf-8 -*-

File: solutions/tests/test_average_of_integers.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+""" "
```
Comment (Alemayehu-Desta) [inline solutions/tests/test_average_of_integers.py:1]: add Python interpreter lines at the top of the file

#!/usr/bin/env python3
# -*- coding: utf-8 -*-

Comment (Alemayehu-Desta) [inline solutions/tests/test_average_of_integers.py:1]: remove the third quote inside the docostring

File: solutions/tests/test_average_of_integers.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        """Test with a list containing non-integer elements."""
+        with self.assertRaises(TypeError):
+            calculate_average([1, 2, "three", 4])
+
+    def test_large_numbers(self):
+        """Test with a list of large numbers."""
+        self.assertEqual(
+            calculate_average([1_000_000, 2_000_000, 3_000_000]), 2_000_000
+        )
+
```
Comment (Alemayehu-Desta) [inline solutions/tests/test_average_of_integers.py:49]: You can addd a single element test

def test_single_element(self):
    """Test with a list containing a single element."""
    self.assertEqual(calculate_average([10]), 10.0)
    
    You can also add all zeros test
    def test_all_zeros(self):
    """Test with a list containing all zeros."""
    self.assertEqual(calculate_average([0, 0, 0, 0]), 0.0)

**Review by Alemayehu-Desta — COMMENTED** [review]
Left few comments for your consideration

**Comment by Mithchell509** [comment]
Thank You for the comment, I added the test suggested.

**Review by Djokwa — APPROVED** [review]

**Review by Papsik09 — APPROVED** [review]
Reviewed everything from my side. Waiting for @Alemayehu-Desta re-review. Nice code and good job!
